src/Symfony/Component/HttpKernel/Debug/FileLinkFormatter.php

```diff
@@ -87,21 +87,16 @@ public static function generateUrlFormat(UrlGeneratorInterface $router, $routeNa
 
     private function getFileLinkFormat()
     {
-        if ($this->fileLinkFormat) {
-            return $this->fileLinkFormat;
-        }
         if ($this->requestStack && $this->baseDir && $this->urlFormat) {
```

Comment:
wouldn't this be called each time now that you removed the first `if` ?

Reply:
Yep, that's on purpose: when you set a request stack, it means the class is used as a service, which means it should be context sensitive. Then, when the stack is empty (eg happens when collecting profiles), we fall back to the last fileLinkFormat.